docs/HowToFAQ/FAQNoBody.md

```diff
@@ -0,0 +1,29 @@
+---
+title: Is there a way to prevent Warning: note, this forall statement has no body  from occurring? I have a forall loop with no body that results in the lemma verifying, but if I add a body (even an empty body) the lemma doesn't verify.
+---
+
+## Question
+
+Is there a way to prevent Warning: note, this forall statement has no body  from occurring? I have a forall loop with no body that results in the lemma verifying, but if I add a body (even an empty body) the lemma doesn't verify.
+
+## Answer
+
+The Dafny verifier allows you to omit the body of various constructs. For example, if you write a method without a body, the verifier will (check the well-formedness of the method specification, and will then) gladly check calls to the method against that specification. This is nice, because it lets you write the specification of a method and then start using it in other places, only to later return to filling in the body of the method.
```

Comment:
```suggestion
The Dafny verifier allows you to omit the body of various constructs. For example, if you write a method without a body, the verifier will gladly check calls to the method against that specification (after checking the well-formedness of the method specification). This is nice, because it lets you write the specification of a method and then start using it in other places, only to later return to filling in the body of the method.
```